internal_configure.bzl

```diff
@@ -18,12 +19,23 @@ def _internal_configure_extension_impl(module_ctx):
     # The pybind11_bazel version should typically just be the pybind11 version,
     # but can end with ".bzl.<N>" if the Bazel plumbing was updated separately.
     version = version.split(".bzl.")[0]
+    integrity = _INTEGRITIES.get(version)
     http_archive(
         name = "pybind11",
         build_file = "//:pybind11-BUILD.bazel",
         strip_prefix = "pybind11-%s" % version,
         url = "https://github.com/pybind/pybind11/archive/refs/tags/v%s.tar.gz" % version,
-        integrity = _INTEGRITIES.get(version),
+        integrity = integrity,
+    )
+
+    metadata_kwargs = {}
+    if bazel_features.external_deps.extension_metadata_has_reproducible:
+        metadata_kwargs["reproducible"] = integrity != None
+
+    return module_ctx.extension_metadata(
+        root_module_direct_deps = ["pybind11"],
```

Comment:
This doesn't seem right?

What this means is the root module, regardless of if it uses pybind or not, _should_ add
```
pybind11_bazel_internal_configure = use_extension("@pybind11_bazel//:internal_configure.bzl", "internal_configure")
use_repo(pybind11_bazel_internal_configure, "pybind")
```

If the root module doesn't add it, a warning will be printed. If the root module, at some point, runs `bazel mod tidy`, it'll be automatically added.

This seems bad as it'll create a public dependency on an internal extension.

Reply:
Sorry if it the case, I'll dig into it.

What's weird is that building from the examples basic module didn't complain about anything.

Reply:
Oh hm. Maybe I misunderstood the docs: https://bazel.build/rules/lib/builtins/module_ctx#extension_metadata

> The names of the repositories that the extension considers to be direct dependencies of the root module. If the root module imports additional repositories or does not import all of these repositories via [use_repo](https://bazel.build/rules/lib/globals/module.html#use_repo), Bazel will print a warning when the extension is evaluated, instructing the user to run bazel mod tidy to fix the use_repo calls automatically.

Maybe there is an implicit "this is only true when the root module uses this extension directly" ?

Reply:
I believe so. I'll ask in the bazel slack.
But in the end we don't need to specify it, I modified the PR accordingly.